opensearch-project/OpenSearch-Dashboards
# [Multiple Datasource] Handle form values(request payload) if the selected type is available in the authentication registry.

### Description


Render credential form for registered from AuthMethod

- “Create DataSource” and "Submit Change" Button enable / disable

### Issues Resolved



Partially fixes https://github.com/opensearch-project/OpenSearch-Dashboards/issues/5692, https://github.com/opensearch-project/OpenSearch-Dashboards/issues/5838

## Screenshot



https://github.com/opensearch-project/OpenSearch-Dashboards/assets/139305463/14bb5e42-73dd-4812-a760-be4c4931b5ac


## Testing the changes


Test steps in this video:
1. Enable MD
2. Navigate to the DataSource creation page:
    - Ensured that the Registered Auth Option appears under the creation form.
    - Confirmed that the "Create data source" button is initially disabled when some fields are empty.
    - Validated that the "Create data source" button becomes enabled when all fields are filled.
3. Proceed to the DataSource Edit page:
    - Verified that the "Registered Auth" option appears under the edition form.
    - Ensured that the "Save changes" button is initially disabled when some fields are empty.
    - Confirmed that the "Save changes" button becomes enabled when all fields are filled.

Additionally, E2E tests have been conducted, and the request payload has been verified as correct for registered Auth Types.


### Check List
